packages/contracts-bedrock/scripts/DeploySuperchain.s.sol

```diff
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Script } from "forge-std/Script.sol";
+
+import { SuperchainConfig } from "src/L1/SuperchainConfig.sol";
+import { ProtocolVersions, ProtocolVersion } from "src/L1/ProtocolVersions.sol";
+import { ProxyAdmin } from "src/universal/ProxyAdmin.sol";
+import { Proxy } from "src/universal/Proxy.sol";
+
+contract DeploySuperchainInput {
+    struct Roles {
+        address proxyAdminOwner;
+        address protocolVersionsOwner;
+        address guardian;
+    }
+
+    struct Input {
+        Roles roles;
+        bool paused;
+        ProtocolVersion requiredProtocolVersion;
+        ProtocolVersion recommendedProtocolVersion;
+    }
+
+    Input internal input;
+    bool internal inputSet = false;
+
+    // -------- Load inputs --------
+
+    function loadInput(string memory _infile) public {
+        _infile;
+        Input memory parsedInput; // TODO parse the input file.
+        loadInput(parsedInput);
+        require(false, "DeploySuperchainInput: loadInput is not implemented");
+    }
+
+    function loadInput(Input memory _input) public {
+        require(!inputSet, "DeploySuperchainInput: input already set");
+        input = _input;
+    }
+
+    // -------- Getter methods for inputs --------
+
+    function inputs() public view returns (Input memory) {
```

Comment:
Maybe it's less verbose if the inputs are just public state vars, with inferred getters? And a single `check` function, that sanity checks that the inputs are all set?